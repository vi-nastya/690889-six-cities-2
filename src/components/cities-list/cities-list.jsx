import React from "react";
import PropTypes from "prop-types";
import {cityPropTypes} from "../../props-types-validation";

const CitiesList = (props) => {
  const {cities, activeCity, onCityChange} = props;

  return <section className="locations container">
    <ul className="locations__list tabs__list">
      {cities.map((city) => {
        return <li key={`city-` + city.name} className="locations__item" onClick={() => onCityChange(city)}>
          <a className={`locations__item-link tabs__item ${city.name === activeCity.name && `tabs__item--active`}`} href="#">
            <span>{city.name}</span>
          </a>
        </li>;
      })}
    </ul>
  </section>;

};

CitiesList.propTypes = {
  activeCity: cityPropTypes,
  onCityChange: PropTypes.func.isRequired,
  cities: PropTypes.arrayOf(cityPropTypes).isRequired,
};

export {CitiesList};

export default CitiesList;
