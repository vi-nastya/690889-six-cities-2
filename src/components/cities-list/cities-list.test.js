import React from "react";
import renderer from "react-test-renderer";
import {CitiesList} from "./cities-list";

it(`Cities list is rendered correctly after relaunch`, () => {
  const tree = renderer.create(<CitiesList cities={[`Amsterdam`, `Moscow`]} activeCity={`Amsterdam`} changeCityHandler={jest.fn()}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
