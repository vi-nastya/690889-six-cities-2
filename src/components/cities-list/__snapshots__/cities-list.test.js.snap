// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities list is rendered correctly after relaunch 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <span
        className="locations__item-link tabs__item tabs__item--active"
      >
        Amsterdam
      </span>
    </li>
    <li
      className="locations__item"
      onClick={[Function]}
    >
      <span
        className="locations__item-link tabs__item false"
      >
        Moscow
      </span>
    </li>
  </ul>
</section>
`;
